plugins/download_statistics/plugin_tests/download_statistics_test.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+
+import os
+import json
+
+from tests import base
+from girder.constants import ROOT_DIR
+
+
+def setUpModule():
+    base.enabledPlugins.append('download_statistics')
+    base.startServer()
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+class DownloadStatisticsTestCase(base.TestCase):
+
+    def setUp(self):
+        base.TestCase.setUp(self)
+
+        # Create admin user
+        admin = {'email': 'admin@email.com',
+                 'login': 'adminLogin',
+                 'firstName': 'adminFirst',
+                 'lastName': 'adminLast',
+                 'password': 'adminPassword',
+                 'admin': True}
+        self.admin = self.model('user').createUser(**admin)
+
+        self.filesDir = os.path.join(ROOT_DIR, 'plugins', 'download_statistics',
+                                     'plugin_tests', 'files')
+
+    def _downloadItem(self, itemId):
+        # Download item through REST api
+        path = '/item/%s/download' % str(itemId)
+        resp = self.request(path, user=self.admin, isJson=False)
+        self.assertStatusOk(resp)
+
+        # Iterate through generator to trigger download events
+        for data in resp.body:
```

Comment:
I'm just curious, do you know why this is necessary? Is the `self.request` somehow lazily downloading files in chunks? From the server's perspective, is more than one request being made?